profile/sandbox/tokenrequest.go

```diff
@@ -0,0 +1,121 @@
+}
+
+// WithDateOfBirth adds a date of birth to the sandbox token request
+func (t TokenRequest) WithDateOfBirth(value time.Time, anchors []Anchor) TokenRequest {
+	formattedTime := value.Format("2006-01-02")
+	return t.WithAttribute(yoti.AttrConstDateOfBirth, formattedTime, anchors)
+}
+
+// WithAgeVerification adds an age-based derivaton attribute to the sandbox token request
+func (t TokenRequest) WithAgeVerification(dateOfBirth time.Time, derivation Derivation, anchors []Anchor) TokenRequest {
+	if anchors == nil {
+		anchors = []Anchor{}
+	}
+	attribute := Attribute{
+		Name:       yoti.AttrConstDateOfBirth,
+		Value:      dateOfBirth.Format("2006-01-02"),
+		Derivation: derivation.ToString(),
+		Anchors:    anchors,
+	}
+	t.Attributes = append(t.Attributes, attribute)
+	return t
+}
+
+// WithGender adds a gender to the sandbox token request
+func (t TokenRequest) WithGender(value string, anchors []Anchor) TokenRequest {
+	return t.WithAttribute(yoti.AttrConstGender, value, anchors)
+}
+
+// WithPhoneNumber adds a phone number to the sandbox token request
+func (t TokenRequest) WithPhoneNumber(value string, anchors []Anchor) TokenRequest {
+	return t.WithAttribute(yoti.AttrConstMobileNumber, value, anchors)
+}
+
+// WithNationality adds a nationality to the sandbox token request
+func (t TokenRequest) WithNationality(value string, anchors []Anchor) TokenRequest {
+	return t.WithAttribute(yoti.AttrConstNationality, value, anchors)
+}
+
+// WithPostalAddress adds a formatted address to the sandbox token request
+func (t TokenRequest) WithPostalAddress(value string, anchors []Anchor) TokenRequest {
+	return t.WithAttribute(yoti.AttrConstAddress, value, anchors)
+}
+
+// WithStructuredPostalAddress adds a JSON address to the sandbox token request
+func (t TokenRequest) WithStructuredPostalAddress(value map[string]interface{}, anchors []Anchor) TokenRequest {
+	data, _ := json.Marshal(value)
+	return t.WithAttribute(yoti.AttrConstStructuredPostalAddress, string(data), anchors)
+}
+
+// WithSelfie adds a selfie image to the sandbox token request
+func (t TokenRequest) WithSelfie(value []byte, anchors []Anchor) TokenRequest {
+	return t.WithBase64Selfie(base64.StdEncoding.EncodeToString(value), anchors)
+}
+
+// WithBase64Selfie adds a base 64 selfie image to the sandbox token request
+func (t TokenRequest) WithBase64Selfie(base64Value string, anchors []Anchor) TokenRequest {
+	return t.WithAttribute(
+		yoti.AttrConstSelfie,
+		base64Value,
+		anchors,
+	)
+}
+
+// WithEmailAddress adds an email address to the sandbox token request
+func (t TokenRequest) WithEmailAddress(value string, anchors []Anchor) TokenRequest {
+	return t.WithAttribute(yoti.AttrConstEmailAddress, value, anchors)
+}
+
+// WithDocumentDetails adds a document details string to the sandbox token request
+func (t TokenRequest) WithDocumentDetails(value string, anchors []Anchor) TokenRequest {
+	return t.WithAttribute(yoti.AttrConstDocumentDetails, value, anchors)
```

Comment:
In some other SDKs we set document details to be optional by default (but not all of them) e.g. https://github.com/getyoti/yoti-dotnet-sdk-sandbox/blob/master/Yoti.Auth.Sandbox/Profile/Request/YotiTokenRequestBuilder.cs#L224

Reply:
Have created SDK-1515 to remove this in other SDKs 👍